crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -157,7 +218,9 @@ impl SettlementEncoder {
                 // have the same price (i.e. are equivalent).
                 return Ok(());
             }
-            (None, None) => bail!("tokens not part of solution for equivalency"),
+            // If token_b is only the buy token of a liquidity order,
+            // the price does not need to be in the uniform clearing prices
+            (None, None) => return Ok(()),
```

Comment:
If what the comment says is true then why are we adding a token equivalency? Shouldn't it still error? Maybe the correct way to handle this is to not call add_token_equivalency for liquidity orders because they shouldn't be buying the native token?